fe/fe-core/src/main/java/org/apache/doris/nereids/trees/TreeNode.java

```diff
@@ -46,4 +48,27 @@
 
     NODE_TYPE withChildren(List<NODE_TYPE> children);
 
+    default void foreach(Consumer<TreeNode<NODE_TYPE>> func) {
```

Comment:
this is a top-down foreach, so maybe it is better to add traverse order info into function name

Reply:
I think we have a promise: if the traverse function no order info, it must be top-down implicitly